### PR TITLE
refactor: extract `op-reth` binary to separate crate

### PR DESCRIPTION
**Pull request by klkvr**
ref https://github.com/paradigmxyz/reth/pull/9439

I've changed `OpChainSpecParser` to return an inner `ChainSpec` for now to avoid introducing more complex bounds on cli types. We can change it back once we figure out how the bounds should look like on underlying types

### CODE REVIEW

**Review by emhane — APPROVED** [review]
great job

**Comment by emhane** [comment]
do mind trying to fix the commit history on this branch? usually does it by checking out a backup branch of your diff from main, then on this branch `git reset --hard main`, then cherry pick from backup branch

**Comment by klkvr** [comment]
> do mind trying to fix the commit history on this branch? usually does it by checking out a backup branch of your diff from main, then on this branch `git reset --hard main`, then cherry pick from backup branch

you mean squashing the diff into less number of commits without merge commits?

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -71,7 +71,7 @@ jobs:
       - uses: Swatinem/rust-cache@v2
         with:
           cache-on-failure: true
-      - run: cargo hack check
+      - run: cargo hack check --workspace --exclude op-reth
```
Comment (emhane) [inline .github/workflows/lint.yml:74]: think you will have to add a new job for op-reth, we still want to check lint for op code

**Review by emhane — DISMISSED** [review]
as long as the diff in 'Files changed' isn't showing unrelated changes, and other people are comfortable reviewing (@joshieDo), then I'm fine with unrelated changes showing in GitHub commit history for the pr

File: .github/workflows/lint.yml (hunk 1 above)
Comment (klkvr) [reply .github/workflows/lint.yml:74]: added a step to run `cargo check` on it with optimism feature

**Review by mattsse — APPROVED** [review]
hell yeah, let's try this again hehe

**Pull request merged**